packages/cache/__tests__/hydration.js

```diff
@@ -35,3 +35,145 @@ test('rehydrated styles to head can be flushed', () => {
   cache.sheet.flush()
   expect(document.documentElement).toMatchSnapshot()
 })
+
+test('flushing rehydrated styles in the head only affect styles matching the cache key', () => {
+  const css = `color:hotpink;`
+  const hash = hashString(css)
+  safeQuerySelector('head').innerHTML = [
+    '<style data-emotion="emo 1lrxbo5">.emo-1lrxbo5{color:hotpink;}</style>',
+    '<style data-emotion="css qweqwee">.css-qweqwee{color:red;}</style>'
+  ].join('')
+
+  // this moves emotion style tags at initialization time
+  jest.resetModules()
+  require('@emotion/react')
+
+  const cache = createCache({ key: 'emo' })
+  expect(document.documentElement).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="css qweqwee"
+          data-s=""
+        >
+          .css-qweqwee{color:red;}
+        </style>
+        <style
+          data-emotion="emo 1lrxbo5"
+          data-s=""
+        >
+          .emo-1lrxbo5{color:hotpink;}
+        </style>
+      </head>
+      <body />
+    </html>
+  `)
+
+  cache.sheet.flush()
+  expect(document.documentElement).toMatchInlineSnapshot(`
+    <html>
+      <head>
+        <style
+          data-emotion="css qweqwee"
+          data-s=""
+        >
+          .css-qweqwee{color:red;}
+        </style>
+      </head>
+      <body />
+    </html>
+  `)
+})
+
+test('Should only hydrate style elements matching the cache key', () => {
+  let css = `color:hotpink;`
+  let hash = hashString(css)
+
+  safeQuerySelector(
+    'body'
+  ).innerHTML = `<style data-emotion="emo ${hash}">.css-${hash}{${css}}</style>`
+
+  const cache = createCache({ key: 'css' })
```

Comment:
`"css"` is reserved for the default cache and we discourage using a custom cache with such key - so maybe worth using a different key here?